### PR TITLE
Issue 127 nicer email template

### PR DESCRIPTION
**Pull request by tsara27**
# Description 

I added a new styled mailer template to Ifme. It's using premailer to render the external css.

## Corresponding Issue

https://github.com/ifmeorg/ifme/issues/127

# Screenshots

screen shot 2018-12-23 at 10 06 23

### CODE REVIEW

File: app/assets/stylesheets/application/mailer.scss [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  padding: 30px;
+  text-align: center;
+
+  &.grid {
+    grid-template-columns: 40% 20% 40%;
+  }
+
+  #footer-content {
+    grid-column-start: 2;
+    color: #A6A5A6;
```
Comment (julianguyen) [inline app/assets/stylesheets/application/mailer.scss]: Let's define this colour in colors.scss or use `$light-grey` instead.

File: app/mailers/application_mailer.rb [hunk 2]
```diff
@@ -104,6 +106,12 @@ def group_notify(data, recipient)
   end
   # rubocop:enable MethodLength
 
+  protected
+
+  def load_logo_inline
+    attachments.inline['logo@2x.png'] = File.read('./public/logo@2x.png')
```
Comment (julianguyen) [inline app/mailers/application_mailer.rb]: Let's unit test this behaviour.

File: app/views/shared/_mailer_header.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<div id="header-container" class="grid">
+  <div id="header-content" class="grid">
+    <%= image_tag(attachments['logo@2x.png'].url) %>
+    <h2 id="header-title">a community for mental health experiences</h2>
```
Comment (julianguyen) [inline app/views/shared/_mailer_header.html.erb]: We should be using i18n here, let's use `<h2 id="header-title">- <%= t('app_description') %></h2>`.

File: app/views/shared/_mailer_header.html.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<div id="header-container" class="grid">
+  <div id="header-content" class="grid">
+    <%= image_tag(attachments['logo@2x.png'].url) %>
```
Comment (julianguyen) [inline app/views/shared/_mailer_header.html.erb]: We should have an alt tag here.

File: app/views/layouts/mailer.html.erb [hunk 5]
```diff
@@ -1,21 +1,17 @@
...
-      <%= t('layouts.mailer.html.no_reply',
-        email: link_to(t('email'), "mailto:join.ifme@gmail.com")
-      ).html_safe %>
-    </p>
+    <%= render 'shared/mailer_header' %>
+    <div class="grid" id="mailer-container">
+      <div id="mailer-content">
+        <%= yield %>
+        <p>
+        -----
```
Comment (nshki) [inline app/views/layouts/mailer.html.erb]: HmmmMmMmMMmm this is just my opinion but let's leave this out so it's closer to the [quick mock](https://user-images.githubusercontent.com/1121087/48811066-097d8400-ecfa-11e8-9c75-29f0ea38a3bd.png). Open to discussion though!

**Comment by julianguyen** [comment]
Any updates here @tsara27? :)

**Comment by tsara27** [comment]
@julianguyen I am sorry Julia. I had tough times at work for few weeks.
I am going to work on this tomorrow.

**Comment by julianguyen** [comment]
No worries at all, I hope you're doing decent!

File: app/views/shared/_mailer_header.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<div class="header-container grid">
+  <div class="header-content grid">
+    <%= image_tag(attachments['logo@2x.png'].url, alt: 'Ifme Logo') %>
```
Comment (julianguyen) [inline app/views/shared/_mailer_header.html.erb]: This alt text should also be an i18n key.

**Comment by julianguyen** [comment]
Any updates? :)

**Comment by julianguyen** [comment]
Checking in on this!

**Pull request merged**